add missing .

babangsund opened the pull request:
Oops. https://github.com/facebook/relay/pull/2846#issuecomment-533221917

mrtnzlml commented:
OK, you did it faster, thanks. I'd personally add the test though since it's a fix of a fix. 😅

sibelius commented:
can you add a test?

babangsund commented:
Probably should. :-)

babangsund commented:
> Probably should. - although uncertain exactly what you want to test?

Well, here's a tiny test for `createPrintRequireModuleDependency` 🤷‍♂

sibelius commented on packages/relay-compiler/codegen/RelayFileWriter.js:
```diff
@@ -24,7 +24,7 @@ const graphql = require('graphql');
 const nullthrows = require('nullthrows');
 const path = require('path');
-const writeRelayGeneratedFile = require('./writeRelayGeneratedFile');
+const {writeRelayGeneratedFile} = require('./writeRelayGeneratedFile');
```
why this?

babangsund replied:
So I could export the function we want to test 🤔

sorenhoyer commented:
After this is merged, would it be possible to make a patch version and do an npm publish, so we don't have to build relay-compiler from master?

mrtnzlml commented:
@sorenhoyer I think it still doesn't work anyway. We have to wait a bit more. I tried multiple times in the past but seems like refetch container still requires the last argument (which makes this unusable - at least for OSS).

sibelius commented:
@mrtnzlml refetch on typescript is broken because of this https://github.com/facebook/relay/issues/2874

mrtnzlml commented:
@sibelius Mm, so it won't work for custom compilers using ESM as well, right? (I use ESM generated output with Flow)

sibelius commented:
after this https://github.com/facebook/relay/pull/2875 it will

alunyov commented on packages/relay-compiler/codegen/RelayFileWriter.js:
```diff
@@ -23,7 +23,7 @@ const graphql = require('graphql');
 const nullthrows = require('nullthrows');
 const path = require('path');
-const writeRelayGeneratedFile = require('./writeRelayGeneratedFile');
+const {writeRelayGeneratedFile} = require('./writeRelayGeneratedFile');
```
Yeah, this, unfortunately, won't work with our internal FB compiler setup. 

So we should probably keep default export from this `writeRelayGeneratedFile`.

alunyov commented on packages/relay-compiler/codegen/__tests__/writeRelayGeneratedFile-test.js:
```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const {
+  createPrintRequireModuleDependency,
```
We should probably make `createPrintRequireModuleDependency` a standalone module, and import it to `writeRelayGeneratedFile`.

babangsund replied:
This can fortunately be done in a jiffy. :-)

alunyov commented on packages/relay-compiler/codegen/writeRelayGeneratedFile.js:
```diff
@@ -169,4 +169,7 @@ function extractHash(text: ?string): ?string {
+module.exports = {
+  createPrintRequireModuleDependency,
+  writeRelayGeneratedFile,
+};
```
Let's keep the old version of the export

babangsund commented:
> Thanks for adding the tests!

My pleasure.

I've moved `createPrintRequireModuleDependency` into a standalone module, renamed the test, and reverted the import/exports.
